File: c/读入一个正整数 n，计算其各位数字之和，用汉语拼音写出和的每一位数字.c
#include<stdio.h>
#define N 100000
int sum(int x){
	if(x==0)
	{printf(" ling");}
	else if(x==1)
	{printf(" yi");}
	else if(x==2)
	{printf(" er");}
	else if(x==3)
	{printf(" san");}
	else if(x==4)
	{printf(" si");}
	else if(x==5)
	{printf(" wu");}
	else if(x==6)
	{printf(" liu");}
	else if(x==7)
	{printf(" qi");}
	else if(x==8)
	{printf(" ba");}
	else if(x==9)
	{printf(" jiu");}
} 
int main(){
	unsigned long a;int d=0,i=1;
	int b=1;
	int c[N];
	scanf("%d",&a);
	while(a>0)
	{
	    b=a%10;
		a=a/10;
		d+=b; 
	}
	printf("%d",d); 
   while(d>0)
   {
   	b=d%10;
   	d=d/10;
   	c[i]=b;
   	i++;
                
}
   while(i>=1)
  {
  	sum(c[i]);
  	i--;
  } 
}

File: python/读入一个正整数 n，计算其各位数字之和，用汉语拼音写出和的每一位数字.py
def sum (a) :
    if a==0 :
       print("ling",end="")
    elif a==1 :
        print("yi",end="")
    elif a==2 :
        print("er",end="")
    elif a==3 :
         print("san",end="")
    elif a==4 :
        print("si",end="")        
    elif a==5 :
         print("wu",end="")
    elif a==6 :
         print("liu",end="")
    elif a==7 :
         print("qi",end="")        
    elif a==8 :
        print("ba",end="")
    elif a==9 :
         print("jiu",end="")

b=int(input())
c=0
while b>0 :
    c=c+b%10
    b=b//10
d=[]
i=0
while c>=1 :
    d.append(c%10)
    c=c//10
    i=i+1
while i>=1 :
    a=d[i-1]
    sum(a)
    if i!=1 :
        print(" ",end="")
    i=i-1
print()